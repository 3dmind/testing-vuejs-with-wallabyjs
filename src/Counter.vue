<template>
  <div>
    <p>
      Counter: {{count}}
    </p>
  </div>
</template>

<script>
  export default {
    name: 'Counter',

    data () {
      return {
        count: 0,
      }
    },

    computed: {
      className () {
        return this.count % 2 === 0
          ? 'even'
          : 'odd'
      }
    },

    methods: {
      increment () {
        this.count += 1
      },

      decrement () {
        this.count += -1
      }
    }
  }
</script>

<style scoped>
  .even {
    color: green;
  }

  .odd {
    color: red;
  }
</style>
