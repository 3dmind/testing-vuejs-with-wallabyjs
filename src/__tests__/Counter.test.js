import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Counter from '../Counter'

const localVue = createLocalVue()

describe('Counter', () => {
  test('render', () => {
    expect.assertions(1)

    const wrapper = mount(Counter, {localVue})

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('counter should be initialized', () => {
    expect.assertions(1)

    const wrapper = shallowMount(Counter, {localVue})

    expect(wrapper.vm.count).toBe(0)
  })

  test('increment count', () => {
    expect.assertions(1)
    const wrapper = shallowMount(Counter, {localVue})

    wrapper.vm.increment()

    expect(wrapper.vm.count).toBe(1)
  })

  test('decrement count', () => {
    expect.assertions(1)
    const wrapper = shallowMount(Counter, {
      localVue,
      data () {return {count: 1}}
    })

    wrapper.vm.decrement()

    expect(wrapper.vm.count).toBe(0)
  })
})
