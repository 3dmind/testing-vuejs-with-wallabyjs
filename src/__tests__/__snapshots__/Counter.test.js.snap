// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render 1`] = `
<div>
  <p class="even">
    Counter: 0
  </p>
  <button>+1</button>
  <button>-1</button>
</div>
`;
